views/admin/javascripts/exhibits.js

```diff
@@ -468,4 +475,219 @@ Omeka.ExhibitBuilder = {};
             });
         });
     };
+
+    Omeka.ExhibitBuilder.setUpCoverImageChooser = function (coverImageChooserUrl, itemOptionsUrl) {
+        var coverImagePanel = $('#cover-image-panel');
+        var selected_cover_image_id = $('#cover_image_file_id').val();
+
+        coverImagePanel.dialog({
+            autoOpen: false,
+            modal: true,
+            resizable: false,
+            create: function () {
+                $(this).dialog('widget').draggable('option', {
+                    containment: 'window',
+                    scroll: false
+                });
+            },
+            open: function () {
+                function refreshDialog() {
+                    coverImagePanel.dialog('option', {
+                        width: Math.min($(window).width() - 100, 600),
+                        height: Math.min($(window).height() - 100, 500)
+                    });
+                }
+
+                refreshDialog();
+                $('body').css('overflow', 'hidden');
+                $(window).on('resize.ExhibitBuilder', function () {
+                    refreshDialog();
+                });
+            },
+            beforeClose: function () {
+                $('body').css('overflow', 'inherit');
+                $(window).off('resize.ExhibitBuilder');
+                $('#cover-image-item-options').empty();
+            },
+            dialogClass: 'item-dialog'
+        });
+
+        /**
+         * Use AJAX to load the form for an attachment.
+         */
+        function loadItemOptionsForm(data, ajaxUrl) {
```

Comment:
I refactored the JavaScript so that there's a one shared loadItemOptionsForm() function.